### PR TITLE
hdev-182-lending-market-ui

### PR DESCRIPTION
**Pull request by tracyarciaga**
## Description of Changes
Tailwind implementation of Lending Market UI Designs from [Figma board](https://www.figma.com/file/lqpLD77mM2olJNrFzuD4Ul/HaloDAO-App?node-id=2233%3A0)

[Link to Jira Ticket hdev-182](https://halodao.atlassian.net/browse/HDEV-182)
[Link to Jira Ticket hdev-78](https://halodao.atlassian.net/browse/HDEV-78)
[Link to Jira Ticket hdev-57](https://halodao.atlassian.net/browse/HDEV-57)

### Developer Checklist:

* [ ] I have followed the guidelines in our Contributing document
* [ ] This PR has a corresponding JIRA ticket
* [ ] My branch conforms with our naming convention i.e. `feature/HDEV-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -575,6 +578,18 @@ export const MainMenu = ({ onClick }: MainMenuProps) => {
 
   return (
     <>
+      <MenuItem>
+        <span onClick={onClick}>
+          <ExternalLink
+            id={`swap-nav-link`}
+            href={
+              'https://app.balancer.fi/#/trade/0x70e8de73ce538da2beed35d14187f6959a8eca96/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
```
Comment (schystz) [inline src/components/Header/index.tsx]: Swap should link to our swap page. See line 484

File: src/components/Tailwind/Buttons/BasicButton.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        flex items-center justify-center
+        font-bold
+        text-white
+        py-1.5 w-full
+        rounded
+        bg-primary
+        ${className}
+        ${isEnabled ? 'cursor-pointer' : 'cursor-not-allowed'}
+        ${isEnabled ? 'opacity-100' : 'opacity-50'}
+      `}
```
Comment (schystz) [inline src/components/Tailwind/Buttons/BasicButton.tsx:39]: Would be nice to add a `hover:` class here so there is a UI indication once the user hover the mouse over the button

File: src/pages/Tailwind/LendMarket/PoolRow.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        <div className="col-6 mb-4 md:mb-0">
+          <div className="text-xs font-semibold tracking-widest uppercase md:hidden">Total Earned:</div>
+          <div className="">{formatNumber(pool.earned, NumberFormat.usd)}</div>
+        </div>
+        <div className="col-7 flex flex-col md:flex-row items-center">
+          <BasicButton
+            title="Lend"
+            isEnabled={true}
+            onClick={() => history.push('/lend-market/lend')}
+            className="md:bg-primary-hover mb-2 md:mb-0 md:mr-4"
```
Comment (schystz) [inline src/pages/Tailwind/LendMarket/PoolRow.tsx]: `md:bg-primary-hover` will only be applied from medium devices up, but not for mobile
Screen Shot 2021-10-18 at 1 10 43 PM

I would recommend adding an additional prop though, like `variant`. Then you just specify the variant like this `variant={BasiButtonVariant.dark}` from this file, and have a conditional class in BasicButton:

```
${variant === BasicButtonVariant.dark ? 'bg-primary-hover' : 'bg-primary' }
```

File: src/components/Tailwind/Buttons/BasicButton.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      className={`
+        flex items-center justify-center
+        font-bold
+        text-white
+        py-1.5 w-full
+        rounded
+        bg-primary
+        ${className}
+        ${isEnabled ? 'cursor-pointer' : 'cursor-not-allowed'}
+        ${isEnabled ? 'opacity-100' : 'opacity-50'}
```
Comment (schystz) [inline src/components/Tailwind/Buttons/BasicButton.tsx]: Lines 21-22 can be simplified by: `${isEnabled ? 'cursor-pointer opacity-100' : 'cursor-not-allowed opacity-50'}`

**Review by schystz — COMMENTED** [review]
Initial review. TBC

File: src/pages/Tailwind/LendMarket/index.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import UserBorrowRow from './UserBorrowRow'
+import { HALO, USDT, XSGD } from '../../../constants'
+import { PoolData, UserLendData } from './models/PoolData'
+import { ArrowUpCircle, ArrowDownCircle } from 'react-feather'
+
+const LendMarket = () => {
+  const [showUserReserves, setShowUserReserves] = useState(true)
+
+  const pools: PoolData[] = [
+    {
```
Comment (0xAplki) [inline src/pages/Tailwind/LendMarket/index.tsx]: we can probably put all test data on a separate folder? something like `/fixtures` so:
1. We can easily replace it with the dynamic implementation
2. Whoever does the integration can refer to that data while developing?

**Review by 0xAplki — COMMENTED** [review]
Also, do we add the connect your wallet first message here as well if the user is not connected?

PS. Our vesting page doesnt have it as well

File: src/pages/Tailwind/LendMarket/index.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  ]
+
+  return (
+    <PageWrapper className="mb-8">
+      <div className="flex flex-col md:flex-row md:space-x-4 pb-2">
+        <div className="w-full md:w-3/6">
+          <PageHeaderLeft
+            subtitle="Overview"
+            title="Lending Markets"
+            caption="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Enim potenti faucibus adipiscing sed tempor diam, ipsum porta."
```
Comment (bitcoinbrisbane) [inline src/pages/Tailwind/LendMarket/index.tsx:25]: Probably need to remove the latin

File: src/pages/Tailwind/LendMarket/UserBorrowRow.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        md:mr-2
+      `}
+    >
+      <Wrapper
+        className={`
+          md:flex md:flex-row md:justify-between md:cursor-pointer md:items-start
+        `}
+      >
+        <div className="col-1 mb-4 md:mb-0">
+          <div className="text-xs font-semibold tracking-widest uppercase md:hidden">Your Borrows:</div>
```
Comment (bitcoinbrisbane) [inline src/pages/Tailwind/LendMarket/UserBorrowRow.tsx:56]: Not sure "Your Borrows" is correct.

Comment (tracyarciaga) [reply src/pages/Tailwind/LendMarket/UserBorrowRow.tsx:56]: It's the same in Aave though @bitcoinbrisbane , will keep it .

File: src/pages/Tailwind/LendMarket/index.tsx (hunk 6 above)
Comment (tracyarciaga) [reply src/pages/Tailwind/LendMarket/index.tsx:25]: We'll replace it when we have a copy for it.

**Review by schystz — APPROVED** [review]

**Review by 0xAplki — APPROVED** [review]

**Pull request merged**